lib/TPP/Dialect/Xsmm/XsmmUtils.cpp

```diff
@@ -78,6 +78,123 @@ FailureOr<UnaryInfo> getUnaryInfo(Value input, Value output) {
   return unaryInfo;
 }
 
+// Examples:
+// If lower=[c], higher=[a, b, c], [c] reshaped into [1, 1, c].
+// If lower=[b, c], higher=[a, b, c], [b, c] reshaped into [1, b, c].
+// If lower=[a], higher=[a, a], [a] reshaped into [1, a].
+// If lower=[a], target=[a, b, a], [a] reshaped into [1, 1, a].
+// If lower=[], target=[a, b, c], [] reshaped into [1, 1, 1].
+void computeBcastShapeInput(ArrayRef<int64_t> higherRankShape,
```

Comment:
this methods will soon disappear. We used them for TPPs but now that we are moving to linalg we can use affine maps. It is unclear also why you need to recompute this informations. Each XSMM operations as a flag specifying the broadcast for each of its operand.

Reply:
removed